base/steer/FairRun.cxx

```diff
@@ -79,14 +79,29 @@ void FairRun::AddTask(FairTask* t)
 {
   fTask->Add(t);
   fNTasks++;
+  StoreTaskNames(t);
+}
+//_____________________________________________________________________________
+void FairRun::StoreTaskNames(const FairTask* t)
+{
+  if(NULL==t) return;
   fFileHeader->AddTaskClassName(t->ClassName());
+  TList* l=t->GetListOfTasks();
+  if (NULL==l) return;
+  for(TObject* t2: *l )
+  {
+    StoreTaskNames(static_cast<FairTask*>(t2));
```

Comment:
Here you need also to add "fNTasks++" to count the subtasks as the counter in AddTask was only increased by one.